### PR TITLE
Fix world deletion

### LINKED ISSUE #7201: Minetest deletes the contents of outside directories instead of the symlinks that point ot them when deleting worlds
##### Issue type
 - Bug report
 
##### Minetest version
```
0.4.16
```

##### OS / Hardware

Operating system: Debian 9
CPU: Irrelevant

##### Summary
If a symbolic link is present in a world directory and Minetest deletes that world, it'll delete the contents of the symbolically linked directory and not the symlink itself. Because the symbolic link doesn't get deleted, this also prevents the world's directory from being properly deleted, so you end up with a world directory containing only the symbolic link to the freshly-emptied outside directory.

##### Steps to reproduce
0. Start a new world.
1. Symbolically link something into that world's directory for use in multiple worlds (for example, a particular `worldmods` directory).
2. Delete that world, expecting the symbolically-linked directory to still be there for use in other worlds.
3. Check your linked directory to find all the mods (or whatever) you had in the linked directory have been erased with no chance of recovery. The linked directory itself still exists, but the contents do not.

### PR DESCRIPTION
**Pull request by numberZero**
Fix #7201

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -594,31 +594,18 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 /******************************************************************************/
 int ModApiMainMenu::l_delete_world(lua_State *L)
 {
-	int worldidx	= luaL_checkinteger(L,1) -1;
-
+	int world_id = luaL_checkinteger(L,1) - 1;
```
Comment (sfan5) [inline src/script/lua_api/l_mainmenu.cpp]: spacing around `,`

File: src/script/lua_api/l_mainmenu.cpp [hunk 2]
```diff
@@ -594,31 +594,18 @@ int ModApiMainMenu::l_create_world(lua_State *L)
...
-		// Delete files
-		if (!fs::DeletePaths(paths)) {
-			lua_pushstring(L, "Failed to delete world");
-		}
-		else {
-			lua_pushnil(L);
-		}
-	}
-	else {
+	if ((world_id < 0) || (world_id >= (int) worlds.size())) {
```
Comment (sfan5) [inline src/script/lua_api/l_mainmenu.cpp]: useless braces

File: src/script/lua_api/l_mainmenu.cpp (hunk 1 above)
Comment (ClobberXD) [reply src/script/lua_api/l_mainmenu.cpp]: Isn't spacing required only to the _right_ of `,`?

Comment (SmallJoker) [reply src/script/lua_api/l_mainmenu.cpp]: Yes. Space *after* `,`

Comment (ClobberXD) [reply src/script/lua_api/l_mainmenu.cpp]: Thank you for clarifying

**Comment by nerzhul** [comment]
interesting legacy code, while we got a recursive removal function :)

**Comment by red-001** [comment]
is #7433 a prerequisite for this?

**Comment by numberZero** [comment]
@red-001 Ugh, forgot that... yes, for Windows™, #7433 is necessary.

File: src/script/lua_api/l_mainmenu.cpp (hunk 1 above)
Comment (numberZero) [reply src/script/lua_api/l_mainmenu.cpp]: added

File: src/script/lua_api/l_mainmenu.cpp (hunk 2 above)
Comment (numberZero) [reply src/script/lua_api/l_mainmenu.cpp]: removed

**Pull request merged**